### ISSUE TITLE
docker run command results in "source is not directory"

### ISSUE THREAD

**Issue by bartgrefte** [eclipse/mosquitto#1656]
Just now I tried to set up a docker image using the docker run command listed at https://hub.docker.com/_/eclipse-mosquitto 
`docker run -it -p 1883:1883 -p 9001:9001 -v mosquitto.conf:/mosquitto/config/mosquitto.conf -v /mosquitto/data -v /mosquitto/log eclipse-mosquitto`

This results in the error "docker: Error response from daemon: source is not directory."
Is there something wrong with that command?

Docker version 19.03.8, build afacb8b7f0
Debian Buster

**Comment by ralight** [comment]
`docker run -it -p 1883:1883 -v <absolute-path-to-configuration-file>:/mosquitto/config/mosquitto.conf eclipse-mosquitto:<version>`

You need to make sure that the path to the configuration file is an absolute path, so `/home/bartgrefte/mosquitto.conf` or wherever it is.

Unless you are using port 9001 in your config there is no need to add in `-p 9001:9001` either.

**Comment by bartgrefte** [comment]
So the command listed at https://hub.docker.com/_/eclipse-mosquitto is incorrect? Or incomplete, depending on how you look at it. 

I just tried to run it again with /home/bart/docker/mosquitto.conf:/mosquitto/config/mosquitto.conf instead of mosquitto.conf:/mosquitto/config/mosquitto.conf, now I'm getting 

docker: Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/home/bart/docker/mosquitto.conf\\\" to rootfs \\\"/var/lib/docker/overlay2/418cc401e3b0c918e01eef1d0416e5840e1ca206a5cedf392ff7caba3e1f24ad/merged\\\" at \\\"/var/lib/docker/overlay2/418cc401e3b0c918e01eef1d0416e5840e1ca206a5cedf392ff7caba3e1f24ad/merged/mosquitto/config/mosquitto.conf\\\" caused \\\"not a directory\\\"\"": unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type.

The folder exists, but the mosquitto.conf does not since this is a clean install. Shouldn't it be created automatically?

**Comment by ralight** [comment]
Yes... it is incomplete. I don't control that page directly but I'll see about getting it fixed.

The local file on your system will not be automatically created. When you use `-v /home/bart/docker/mosquitto.conf:/mosquitto/config/mosquitto.conf` you are saying "please mount the already existing file at /home/bart/docker/mosquitto.conf over the top of the file /mosquitto/config/mosquitto.conf inside the image, so my local file is used as the configuration instead".

If you just want to run the image with the default configuration then do:

`docker run -it -p 1883:1883 eclipse-mosquitto`

I can see that the documentation on dockerhub assumes a lot of knowledge, that should be improved.

**Comment by bartgrefte** [comment]
Ah, that clears things up, thanks :)
